File: app/src/main/java/com/robifr/ledger/ui/searchproduct/recycler/SearchProductHeaderHolder.java
/**
 * Copyright (c) 2024 Robi
 *
 * Ledger is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Ledger is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Ledger. If not, see <https://www.gnu.org/licenses/>.
 */

package com.robifr.ledger.ui.searchproduct.recycler;

import androidx.annotation.NonNull;
import androidx.core.text.HtmlCompat;
import com.robifr.ledger.R;
import com.robifr.ledger.data.model.ProductModel;
import com.robifr.ledger.databinding.ListableListTextBinding;
import com.robifr.ledger.ui.RecyclerViewHolder;
import com.robifr.ledger.ui.searchproduct.SearchProductFragment;
import java.util.List;
import java.util.Objects;
import java.util.Optional;

public class SearchProductHeaderHolder extends RecyclerViewHolder<Optional> {
  @NonNull private final SearchProductFragment _fragment;
  @NonNull private final ListableListTextBinding _textBinding;

  public SearchProductHeaderHolder(
      @NonNull SearchProductFragment fragment, @NonNull ListableListTextBinding binding) {
    super(binding.getRoot());
    this._fragment = Objects.requireNonNull(fragment);
    this._textBinding = Objects.requireNonNull(binding);
  }

  @Override
  public void bind(@NonNull Optional ignore) {
    final List<ProductModel> products =
        this._fragment.searchProductViewModel().products().getValue();
    final int totalProducts = products != null ? products.size() : 0;
    final String text =
        this._fragment
            .getResources()
            .getQuantityString(R.plurals.args_found_x_product, totalProducts, totalProducts);

    this._textBinding.text.setText(HtmlCompat.fromHtml(text, HtmlCompat.FROM_HTML_MODE_LEGACY));
  }
}

File: app/src/main/java/com/robifr/ledger/ui/searchproduct/recycler/SearchProductHeaderHolder.kt
/**
 * Copyright 2024 Robi
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.robifr.ledger.ui.searchproduct.recycler

import androidx.core.text.HtmlCompat
import com.robifr.ledger.R
import com.robifr.ledger.data.model.ProductModel
import com.robifr.ledger.databinding.ListableListTextBinding
import com.robifr.ledger.ui.RecyclerViewHolderKt

class SearchProductHeaderHolder(
    private val _textBinding: ListableListTextBinding,
    private val _products: () -> List<ProductModel>
) : RecyclerViewHolderKt(_textBinding.root) {
  override fun bind(itemIndex: Int) {
    _textBinding.text.text =
        HtmlCompat.fromHtml(
            itemView.context.resources.getQuantityString(
                R.plurals.searchProduct_found_x_product, _products().size, _products().size),
            HtmlCompat.FROM_HTML_MODE_LEGACY)
  }
}
